### PR TITLE
Issue #1438, refactored Api.err

### PR DESCRIPTION
**Pull request by michael-volynets**
Issue #1438,
Refactored Api.Err

### CODE REVIEW

File: extension/js/common/api/error/api-error.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
+import { Xss } from '../../platform/xss.js';
+import { AjaxErr, ApiErrResponse, StandardError, StandardErrRes, AuthErr, GoogleAuthErr } from './api-error-types.js';
+import { Catch } from '../../platform/catch.js';
+
+export class ApiErr {
+    static eli5(e: any): string {
```
Comment (tomholub) [inline extension/js/common/api/error/api-error.ts]: Please use fat arrows everywhere :) `static eli5 = (e: any): string => {`

Then it should not complain about unbound methods when you do `.catch(ApiErr.reportIfSignificant)`

**Comment by tomholub** [comment]
> Yes, this is exactly what I wanted. Just use the fat arrow prop methods. Actually the code style tests should fail on this too if you forked from master (I hope).

The tests did not fail because the regexes assume two-space indents. You seem to be using 4 spaces, please check your vscode config.

File: extension/js/common/api/error/api-error.ts (hunk 1 above)
Comment (michael-volynets) [reply extension/js/common/api/error/api-error.ts]: Ohh, my bad
Will change

**Comment by tomholub** [comment]
> > Yes, this is exactly what I wanted. Just use the fat arrow prop methods. Actually the code style tests should fail on this too if you forked from master (I hope).
> 
> The tests did not fail because the regexes assume two-space indents. You seem to be using 4 spaces, please check your vscode config.

I went ahead and committed the config to the workspace, so you can just pull from master to get the settings: https://github.com/FlowCrypt/flowcrypt-browser/commit/e7f87341a2bbb4fca9215c8627711f7a29d40181

**Comment by michael-volynets** [comment]
> I went ahead and committed the config to the workspace, so you can just pull from master to get the settings: e7f8734
Thanks

**Comment by tomholub** [comment]
Also please undo this commit, it should be exactly left with the arrow functions, everywhere: https://github.com/FlowCrypt/flowcrypt-browser/pull/2328/commits/207e9ebb5198e8d1259744a5b024ff85e53c01e3

**Comment by michael-volynets** [comment]
Time Spent: 1h 40m

**Comment by michael-volynets** [comment]
Spaces haven't changed, let me check

File: extension/chrome/settings/index.ts [hunk 2]
```diff
@@ -214,7 +214,7 @@ View.run(class SettingsView extends View {
         const html = `<div class="line"><a href="https://flowcrypt.com${Xss.escape(post.url)}" target="_blank">${Xss.escape(post.title.trim())}</a> ${Xss.escape(post.date.trim())}</div>`;
         Xss.sanitizeAppend('.blog_post_list', html);
       }
-    }).catch(e => ApiErr.isSignificant(e) ? Catch.reportErr(e) : undefined);
+    }).catch(e => ApiErr.isSignificant(e) && Catch.reportErr(e));
```
Comment (tomholub) [inline extension/chrome/settings/index.ts]: Can replace with `}).catch(ApiErr.reportIfSignificant);`

Comment (michael-volynets) [reply extension/chrome/settings/index.ts]: Thanks

File: extension/js/common/api/error/api-error.ts [hunk 3]
```diff
@@ -126,10 +126,10 @@ export class ApiErr {
         return e instanceof AjaxErr && e.status === 403 && e.responseText.indexOf('insufficientPermissions') !== -1;
     }
     static isNotFound = (e: any): e is AjaxErr => e instanceof AjaxErr && e.status === 404;
-    static isReqTooLarge(e: any): boolean { return e instanceof AjaxErr && e.status === 413; }
-    static isServerErr(e: any): boolean { return e instanceof AjaxErr && e.status >= 500; }
+    static isReqTooLarge = (e: any): boolean => e instanceof AjaxErr && e.status === 413;
+    static isServerErr = (e: any): boolean => e instanceof AjaxErr && e.status >= 500;
```
Comment (tomholub) [inline extension/js/common/api/error/api-error.ts]: I'm also deprecating all one-line methods with implicit return, for consistency. So all should be written as 

```js
  static isServerErr = (e: any): boolean => {
    return e instanceof AjaxErr && e.status >= 500;
  }
```

It's more code but it will be better for consistency, overall readability should increase when it's consistent.

File: extension/chrome/settings/index.ts (hunk 2 above)
Comment (tomholub) [inline extension/chrome/settings/index.ts]: Can replace with `}).catch(ApiErr.reportIfSignificant);`

**Comment by tomholub** [comment]
> Spaces haven't changed, let me check

Try to restart vscode. I also have the same setting in my personal settings in vscode, take a look there too.

File: extension/chrome/settings/index.ts (hunk 2 above)
Comment (michael-volynets) [reply extension/chrome/settings/index.ts]: Thanks!

File: extension/js/common/api/error/api-error.ts (hunk 3 above)
Comment (michael-volynets) [reply extension/js/common/api/error/api-error.ts]: Okay, got it

**Comment by tomholub** [comment]
@michael-volynets there seem to be many failing tests, does not look like flakes: https://flowcrypt.semaphoreci.com/jobs/9c5445b0-3b75-4aef-ad4d-ade10f87f85b

**Comment by michael-volynets** [comment]
> @michael-volynets there seem to be many failing tests, does not look like flakes: https://flowcrypt.semaphoreci.com/jobs/9c5445b0-3b75-4aef-ad4d-ade10f87f85b

Thanks, I've started working on Pgp.armor so I didn't notice it.
I will take a look

**Comment by michael-volynets** [comment]
Found the problem, fixing
